src/Types/ListType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace ILIAS\Types;
+
+/**
+ * Class ListType
+ *
+ * @package ILIAS\Types
+ *
+ * @author  Oskar Truffer <ot@studer-raimann.ch>
+ *
+ * Describes a list of a certain Type.
+ *
+ * Attention:
+ * isExtensionOf behaves Covariant!
+ * new ListType(Dog::class).isExtensionOf(new ListType(Animal::class)) == true!
+ *
+ * See: https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)
+ */
+class ListType implements Type, Ancestors {
+
+	/** @var Type */
+	protected $type;
+
+
+	/**
+	 * SingleType constructor.
+	 *
+	 * @param $fullyQualifiedClassName string|Type Give a Value Type or a Type that will be wrapped in a single type.
+	 */
+	public function __construct($fullyQualifiedClassName) {
+		if (!is_a($fullyQualifiedClassName, Type::class)) {
+			$fullyQualifiedClassName = new SingleType($fullyQualifiedClassName);
+		}
+		$this->type = $fullyQualifiedClassName;
+	}
+
+
+	/**
+	 * @inheritdoc
+	 */
+	function __toString() {
+		return "[" . $this->type . "]";
+	}
+
+
+	/**
+	 * Is this type a subtype of $type. Not strict! x->isExtensionOf(x) == true.
+	 *
+	 * Attention: [Dog].isExtensionOf([Animal]) == true. In other words:
+	 * The isExtensionOf behaves Covariant!
+	 *
+	 * If you are familiar with e.g. Java Generics ?.isExtensionOf(x) behaves the same as <? extends x>.
```

Comment:
I partially agree:
>  There is no meaningful way to implement a List of (subclass of X)s as a subclass of List of Xs
Is certainly true, thus I did not call it isSubclassOf but isExtensionOf. The difference is what I tried to explain in the commentary: isSubclassOf (not implemented here) would behave covariant on SingleTypes and contravariant on ListTypes. isExtensionOf behaves covariant in both the SingleType and the List Type. With the one crucial effect:

new ListType(Dog::class).isExtensionOf(new ListType(Animal::class)) == true

Thus it is also valid to implement isExtensionOf on SingleTypes using the reflection class' isSubclass, as on the SingleType they both behave covariant.

Why I think the covariant implementation is more useful atm (we can have both later) than the contravariant, and why I use the covariant in the background tasks (this is open for discussion): 
If I have a job that accepts ILIAS Objects "ListType(ILIASObject)" and gives back their ref ids "ListType(IntegerValue)" I want the Job to also be able to handle "ListType(ilObjCourse)", as getting the ref ids of different ilObjects would be tedious, thus we need the covariant implementation somehow. 

To be perfectly in sync with e.g. Java we would need an additional typelike helper Extends(_). Then we could our ref id job to take an argument of type ListType(Extends(SingleType(ilObject))). This would be similar to:

[1] List<Animal>  <--> ListType(SingleType(Animal))
[2] List<? extends Animal>  <-->  ListType(Extends(SingleType(Animal)))

Where the isSubclassOf behaves contravariant for [1] and covariant for [2]. We might want to implement isSubclassOf later on, or do you think this is urgent? Thus follows for your question:

> I thus don't see which information I (as a user of Type) get, if I call isExtensionOf on a type object.
You always get covariant behaviour. And yes, this is a concern for type safety.